vitrina/resources/views.py

```diff
@@ -285,28 +281,25 @@ def handle_no_permission(self):
     def form_valid(self, form: BaseForm) -> HttpResponse:
         resource = get_object_or_404(DatasetDistribution, id=self.kwargs["pk"])
         dataset = get_object_or_404(Dataset, id=resource.dataset_id)
-        with create_revision():
-            add_to_revision(resource)
-            set_user(self.request.user)
-            set_comment((f'Ištrintas duomenų šaltinis "{resource.lt_title()}".'))
-            resource.delete()
-
-            if not DatasetDistribution.objects.filter(dataset=dataset) and dataset.is_public:
-                if dataset.plandataset_set.exists():
-                    dataset.status = Dataset.PLANNED
-                    comment_status = Comment.PLANNED
-                else:
-                    dataset.status = Dataset.INVENTORED
-                    comment_status = Comment.INVENTORED
-
-                Comment.objects.create(
-                    content_type=ContentType.objects.get_for_model(dataset),
-                    object_id=dataset.pk,
-                    type=Comment.STATUS,
-                    status=comment_status,
-                    user=self.request.user,
-                )
-                dataset.save()
+        add_to_revision(resource)
+        resource.delete()
+
+        if not DatasetDistribution.objects.filter(dataset=dataset) and dataset.is_public:
```

Comment:
Quicker & less resource heavy check:

```suggestion
        if not DatasetDistribution.objects.filter(dataset=dataset).exists() and dataset.is_public:
```